### ISSUE TITLE
TypeScript erroneously reports unused variable 'toDate'

### ISSUE THREAD

**Issue by wizardnet972** [microsoft/TypeScript#61674]
**Description:**

I'm encountering an issue where TypeScript reports that the variable `toDate` is declared but its value is never read (Error Code 6133). However, the variable `toDate` is actually used, so this seems to be a false positive in TypeScript's analysis.

### Steps to Reproduce:

1. Use the following TypeScript code:

   ```ts
   function foo() {
     let toDate = moment().startOf('day');
     if (true) {
       toDate = moment(new Date(this.datePicker.fromDate)).endOf('day');
     }
   }
   ```

2. The TypeScript compiler will raise the error:

   ```
   'toDate' is declared but its value is never read. (6133)
   ```

### Expected Behavior:

TypeScript should recognize that `toDate` is assigned a new value within the `if` block, and it should not report it as "unused."

### Actual Behavior:

TypeScript reports that `toDate` is declared but never used, even though the variable is reassigned in the `if` block.

https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABMOcAUBKRBvAUIxAGwFMpEo4ARAQymMQF5EBbOZ4sKTAOgGcpqAJygB5YGgDkAE2oBPCRgDc+RDGCI0UQSGJY8BAhRp1GLNhy5hiAd0THimgBYxe3GXQAKMCAGtig7mBBNnsMDG4OKTFJGXklFQBfXASgA

**Comment by wizardnet972** [comment]
After experimenting with it, I noticed that if the variable isn't accessed, ts treated as unused.
